File: lesunny/061_프린터.java
import java.util.LinkedList;

class DocNode{
    int idx;
    int priority;
    public DocNode(int idx, int priority){
        this.idx = idx;
        this.priority = priority;
    }
}

class Solution {
    public int solution(int[] priorities, int location) {
        int answer = 1;
        LinkedList<DocNode> documents = new LinkedList<>();
        
        for(int i=0; i<priorities.length ; i++){
            documents.add(new DocNode(i, priorities[i]));
        }
        
        while(documents.size()>1){
            DocNode firstDoc = documents.getFirst();
            
            for(int i=1; i<documents.size(); i++){
                if(firstDoc.priority < documents.get(i).priority){
                    documents.addLast(firstDoc);
                    documents.removeFirst();
                    break;
                }
                
                //print
                if(i==documents.size()-1){
                    if(firstDoc.idx==location)
                        return answer;
                    else{
                        documents.removeFirst();
                        answer++;
                    }
                }
            }
        }
        
        return answer;
    }
}

File: JIWON/061 프린터.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> priorities, int location) {
    int answer = 0;
    int max = 0;
    
    for(int i=0;i<priorities.size();i++){
        if(max <= priorities[i]) max = priorities[i];
    }
    
    int i = 0;
    while(true){
        if(i == priorities.size()) i = 0;
        
        if(max == priorities[i]){
            answer++;
            max = 0;
            
            if(location == i){
                return answer;
            }
            priorities[i] = -1;
            
            for(int j=0;j<priorities.size();j++){
                if(max <= priorities[j]) max = priorities[j];
            }
        }
        i++;
    }
    
    return answer;
}
